packages/connect-react/src/index.ts

```diff
@@ -184,52 +211,108 @@ export function usePermissions(): [Permission[], LoadingStatus] {
   return useConnectSubscription<Permission[]>(callback, [])
 }
 
-export function createAppHook(appConnect: Function) {
+export function createAppHook(
+  appConnect: Function,
+  connector?: string | [string, { [key: string]: any } | undefined]
+) {
   return function useAppData<T = any>(
     app: App | null,
-    callback?: (app: App | any) => T | Promise<T>
+    callback?: (app: App | any) => T | Promise<T>,
+    dependencies?: any[]
   ): [T | null, LoadingStatus] {
-    const [result, setResult] = useState<T | null>(null)
-    const [error, setError] = useState<Error | null>(null)
-    const [loading, setLoading] = useState<boolean>(false)
+    const [{ result, error, loading }, setStatus] = useState<{
+      error: Error | null
+      loading: boolean
+      result: T | null
+    }>({
+      error: null,
+      loading: false,
```

Comment:
Should the `loading` statuses be defaulting to `true`? at the moment it's confusing because conditionally accessing data behind these flags on the first render will throw on `undefined / null`.

Reply:
Yes you’re right! I updated the default loading states in https://github.com/aragon/connect/pull/233/commits/add4da4ead869eee868ea66d870a3a5a95353a22

Reply:
Amazing, to use the same example as before: `orgStatus` works great now! but I'm still seeing the same issue with `votesStatus` via `useVoting` though.

Reply:
@andy-hook Trying to reproduce but I’m not sure how to do it, is the issue that `votes` can be `null`? If that’s the case, this is intended since we don’t know in advance the type of data here, so we can’t really have a default other than `null`. A solution I can see could be to have another optional param to pass an initial value, maybe after the dependencies. If the issue is not this, could you post a way to reproduce it? Thanks!

Reply:
Sorry if I wasn't clear. I was referring to the `loading` status when using `useVoting`. It reports as `false` initially, even though the votes have yet to be loaded.

Reply:
Oh sorry I misunderstood, it was clear :smile: Let me check for the `loading` status.

Also this gave me the idea of returning `undefined` rather than `null` for the values, so that we can use the default value of the destructuring assignment:

```js
// `votes` is always an array
const [votes = [], votesStatus] = useVoting(voting, app => app.onVotes())
```

Reply:
@andy-hook It should work now, let me know if you still have the issue!